lms/djangoapps/badges/tests/test_models.py

```diff
@@ -105,6 +106,19 @@ def test_unique_for_course(self):
         self.assertNotEqual(badge_class.id, course_badge_class.id)
         self.assertEqual(course_badge_class.id, premade_badge_class.id)
 
+    def test_get_badge_class_course_disabled(self):
+        """
+        Verify attempting to fetch a badge class for a course which does not issue badges raises an
+        exception.
+        """
+        course_key = CourseFactory.create(metadata={'issue_badges': False}).location.course_key
+        self.assertRaises(
```

Comment:
Nit: Use this as a context manager (`with self.assertRaises(CourseBadgesDisabledError):`) and the syntax is way nicer.